source/math.d.ts

```diff
@@ -0,0 +1,224 @@
+import type {NumberAbsolute, BuildTuple, And, Or, ArrayMax, ArrayMin, PositiveNumericStringGt} from './internal';
+import type {IsEqual} from './is-equal';
+import type {PositiveInfinity, NegativeInfinity, IsNegative} from './numeric';
+
+/**
+Returns the sum of the two input numbers.
+
+Note:
+- A or B can only support `-999` ~ `999`.
+- A and B can only be small integers(less than 1000).
+- if the result is negative, you can only get `number`.
+
+@example
+```
+import type {Add} from 'type-fest';
+
+Add<111, 222>;
+//=> 333
+Add<-111, 222>;
+//=> 111
+Add<111, -222>;
+//=> number
+Add<PositiveInfinity, -9999>;
+//=> PositiveInfinity
+Add<PositiveInfinity, NegativeInfinity>;
+//=> number
+```
+
+@category Numeric
+*/
+// TODO: Support big integer and negative number.
+export type Add<A extends number, B extends number> = [
```

Comment:
I think this will lead to the `internal.d.ts` getting bigger and harder to maintain, would it be better to put the `math.d.ts` in the `internal` folder?

Reply:
> I think this will lead to the internal.d.ts getting bigger and harder to maintain

The goal is to publicly export many things in internal.d.ts. See my comment in https://github.com/sindresorhus/type-fest/pull/748#discussion_r1515869580

Reply:
> `Add` and `Subtract` should be separate files and the rest of the utilities should be moved into `internal.d.ts`.

fine

Reply:
exported them!